File: Rust/0026-remove-duplicates-from-sorted-array.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut i_read:  usize = 1;
        let mut i_write: usize = 1;

        while i_read < nums.len() {
            if nums[i_read] != nums[i_write - 1] {
                nums[i_write] = nums[i_read];
                i_write += 1;
            }

            i_read += 1;
        }

        return i_write as i32;
    }
}

File: c++/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size() == 0) { return 0; }

        size_t read = 1;
        size_t write = 1;

        while(read < nums.size()) {
            if(nums[read - 1] != nums[read]) {
                if(write != read) {
                    nums[write] = nums[read];
                }
                write += 1;
            }

            read += 1;
        }

        return write;
    }
};
